packages/hop-node/src/watchers/BondWithdrawalWatcher.ts

```diff
@@ -347,6 +369,97 @@ class BondWithdrawalWatcher extends BaseWatcher {
       }
     })
   }
+
+  async preTransactionValidation (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Perform this check as late as possible before the transaction is sent
+    await this.validateDbExistence(txParams)
+    await this.validateTransferSentIndex(txParams)
+    await this.validateUniqueness(txParams)
+    await this.validateLogsWithBackupRpc(txParams)
+  }
+
+  async validateDbExistence (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate DB existence with calculated transferId
+    const calculatedDbTransfer = await this.getCalculatedDbTransfer(txParams)
+    if (calculatedDbTransfer.transferId !== txParams.transferId) {
+      throw new PreTransactionValidationError(`Calculated transferId (${calculatedDbTransfer?.transferId}) does not match transferId in db`)
+    }
+  }
+
+  async validateTransferSentIndex (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate transferSentIndex is expected since it is not part of the transferId
+    const calculatedDbTransfer = await this.getCalculatedDbTransfer(txParams)
+    if (calculatedDbTransfer.transferSentIndex !== txParams.transferSentIndex) {
```

Comment:
suggestion: check that `calculatedDbTransfer.transferSentIndex` is not undefined